emcc.py

```diff
@@ -2286,8 +2286,8 @@ def compile_source_file(i, input_file):
       # unless running in strict mode.
       if not shared.Settings.STRICT and '-nostdlib++' not in newargs:
         link_as_cxx = True
-      extra_files_to_link += system_libs.calculate([f for _, f in sorted(temp_files)] + extra_files_to_link, link_as_cxx, forced=forced_stdlibs)
-    linker_inputs += extra_files_to_link
+      extra_files_to_link += system_libs.calculate([f for _, f in sorted(linker_inputs)] + extra_files_to_link, link_as_cxx, forced=forced_stdlibs)
+    link_flags += extra_files_to_link
```

Comment:
maybe `link_arguments` since it includes both files and flags?

Reply:
SGTM